core/src/main/java/io/questdb/cairo/DatabaseCheckpointAgent.java

```diff
@@ -307,6 +321,17 @@ private void checkpointCreate(SqlExecutionContext executionContext, CharSequence
                                     reader.getColumnVersionReader().dumpTo(mem);
                                     mem.close(false);
 
+                                    if (lightweightCheckpointSupported) {
+                                        long txn = reader.getTxn();
+                                        TxnScoreboard scoreboard = engine.getTxnScoreboard(tableToken);
+                                        if (!scoreboard.acquireTxn(TxnScoreboard.CHECKPOINT_ID, txn)) {
```

Comment:
nit: we could use `incrementTxn()` instead of `acquireTxn()` here and then we can get rid of the `id > -1` checks in V2's `acquireTxn()` impl.

Reply:
changed in [a47b787](https://github.com/questdb/questdb/pull/5444/commits/a47b787faef4b4af9d4f93301f60c2c3ea0af3d9)